src/docs/binding-structs.md

```diff
@@ -0,0 +1,100 @@
+
+Structs in Swift are encouraged to be immutable. To help along with this, the compiler enforces a number of rules to prevent you from trying to mutate an immutable value. All arguments to a method are considered to be `let` bindings. The compiler won't let you modify public fields of struct that is defined with a `let` binding. Similarly, you cannot call a `mutating` method on a struct that is bound with `let`.
+
+Every value type in Swift comes with a compiler-generated Value Witness Table. Unlike other function tables in Swift (Protocol witness tables, vtables etc), the layout of value witness tables are well-defined. The value witness table is documented [here](binding-value-witness-table.md).
+
+## Language Parity Mismatches
+
+Structs in C# and Swift are very similar. They both represent value types with a concrete memory layout. The primary difference is that structs in Swift will execute code whenever it goes out of scope.
+
+Given the following Swift code:
+```swift
+public class Named {
+    private let name: String
+    public init(name n: String)
+    {
+        name = n
+    }
+    deinit
+    {
+        print("you killed \(name).")
+    }
+}
+
+public struct NamedHolder {
+    public init (name n: String) {
+        name = Named(name: n);
+    }
+    public var name: Named
+}
+
+public func runIt()
+{
+    var n1:NamedHolder = NamedHolder(name: "Corey")
+    let n2:NamedHolder = NamedHolder(name: "Ian")
+    n1 = n2
+}
+runIt()
+print("all done")
+```
+It will generate the following output:
+```
+you killed Corey.
+you killed Ian.
+all done
+```
+This shows how the destructor in the class gets executed when a struct gets overwritten ("you killed Corey") and when it goes out of scope. C# will do neither of this things.
+
+Swift allows 0-length structs, whereas C# does not.
+Swift allows structs to be non-copyable by adding the pseudo inheritance `: ~Copyable` to the type declaration. These types may present issues when
+being bound. The value witness table contains functions that will execute an illegal instruction if they are called to copy the type and all mechanisms
+within swift to get the address of an non-copyable instance are forbidden.
+
+## ABI Differences
+Swift has very specific rules for packing structs which Apple has laid out [here](https://github.com/swiftlang/swift/blob/main/docs/ABI/TypeLayout.rst).
+
+Structs are passed to functions in one of two ways depending on whether or not they are frozen under `enable-module-evolution` rules.
+If the struct is frozen, it is lowered into up to 4 registers, otherwise it is passed by reference. This is usually done by copying it onto the stack and taking the address into a register. If the func is an instance method it will be passed by reference in the self register.
```

Comment:
```suggestion
If the struct is frozen and bitwise-copyable, it is lowered into up to 4 registers, otherwise it is passed by reference. This is usually done by copying it onto the stack and taking the address into a register. If the func is an instance method it will be passed by reference in the self register.
```